app/services/offender_service.rb

```diff
@@ -18,6 +18,23 @@ def self.get_offender(offender_no)
     }
   end
 
+  def self.get_community_data(nomis_offender_id)
+    community_info = HmppsApi::CommunityApi::get_offender(nomis_offender_id).deep_symbolize_keys
+    registrations = HmppsApi::CommunityApi::get_offender_registrations(nomis_offender_id).
+        map(&:symbolize_keys).select { |r| r.fetch(:active) && r.fetch(:registerLevel).starts_with?('M') }
```

Comment:
I think you've got the format wrong for the `registerLevel` field.

I have a feeling this implementation (and associated `:community_registration` factory) isn't faithful to the Community API. The value of `registerLevel` doesn't match what I'd expect to see from the Community API.

The factory returns a String:

```json
"registerLevel": "M1"
```

But [the Swagger docs](https://community-api-public.test.delius.probation.hmpps.dsd.io/swagger-ui/index.html#/Registrations/getOffenderRegistrationsByNomsNumberUsingGET) say it should be a Hash:

```json
"registerLevel": {
  "code": "string",
  "description": "string"
}
```

I've hit the real Community API and can confirm that it behaves as documented. Here's a response which is returned for a 'MAPPA level 0' offender:

```json
"registerLevel": {
  "code": "M0",
  "description": "MAPPA Nominal (level to be determined)"
}
```

Reply:
**Also:** in real-world Community API responses, the `registrations` endpoint also returns Hashes which don't include a `registerLevel` key.

It looks like `registerLevel` is optional, so we should treat it as such here and ignore registrations without it.